packages/node/src/methods/install-operations.ts

```diff
@@ -82,8 +104,175 @@ export async function addAppInstance(
   } else {
-    await this.channels.install(params);
+    await install(this.store, params);
+  }
+}
+
+/**
+ * Creates a ProposedAppInstanceInfo to reflect the proposal received from
+ * the client.
+ * @param selfAddress
+ * @param store
+ * @param params
+ */
+async function createProposedAppInstance(
+  selfAddress: Address,
+  store: Store,
+  params: Node.ProposeInstallParams
+): Promise<string> {
+  const appInstanceId = generateUUID();
+  const channel = await getChannelFromPeerAddress(
+    selfAddress,
+    params.peerAddress,
+    store
+  );
+
+  const proposedAppInstance = new ProposedAppInstanceInfo(
+    appInstanceId,
+    params
+  );
+
+  await store.addAppInstanceProposal(channel, proposedAppInstance);
+  return appInstanceId;
+}
+
+async function install(
+  store: Store,
+  params: Node.InstallParams
+): Promise<AppInstanceInfo> {
+  if (!params.appInstanceId) {
+    return Promise.reject("No AppInstanceId specified to install");
+  }
+
+  const stateChannel = await store.getChannelFromAppInstanceID(
+    params.appInstanceId
+  );
+
+  const appInstanceInfo = await store.getProposedAppInstanceInfo(
+    params.appInstanceId
+  );
+  const appInstance: AppInstance = createAppInstanceFromAppInstanceInfo(
+    appInstanceInfo,
+    stateChannel
+  );
+  delete appInstanceInfo.initialState;
+
+  const updatedStateChannel = stateChannel.installApp(
+    appInstance,
+    appInstanceInfo.myDeposit,
+    appInstanceInfo.peerDeposit
+  );
+
+  await store.updateChannelWithAppInstanceInstallation(
+    updatedStateChannel,
+    appInstance,
+    appInstanceInfo
+  );
+
+  return appInstanceInfo;
+}
+
+/**
+ * @param appInstanceInfo The AppInstanceInfo to convert
+ * @param channel The channel the AppInstanceInfo belongs to
+ */
+function createAppInstanceFromAppInstanceInfo(
+  proposedAppInstanceInfo: ProposedAppInstanceInfo,
+  channel: StateChannel
+): AppInstance {
+  const appFunctionSigHashes = getAppFunctionSigHashes(proposedAppInstanceInfo);
+
+  const appInterface: AppInterface = {
+    addr: proposedAppInstanceInfo.appId,
+    applyAction: appFunctionSigHashes.applyAction,
+    resolve: appFunctionSigHashes.resolve,
+    getTurnTaker: appFunctionSigHashes.getTurnTaker,
+    isStateTerminal: appFunctionSigHashes.isStateTerminal,
+    stateEncoding: proposedAppInstanceInfo.abiEncodings.stateEncoding,
+    actionEncoding: proposedAppInstanceInfo.abiEncodings.actionEncoding
+  };
+
+  // TODO: throw if asset type is ETH and token is also set
+  const terms: Terms = {
+    assetType: proposedAppInstanceInfo.asset.assetType,
+    limit: proposedAppInstanceInfo.myDeposit.add(
+      proposedAppInstanceInfo.peerDeposit
+    ),
+    token: proposedAppInstanceInfo.asset.token
+      ? proposedAppInstanceInfo.asset.token
+      : AddressZero
+  };
+
+  return new AppInstance(
+    channel.multisigAddress,
+    // TODO: generate ephemeral app-specific keys
+    channel.multisigOwners,
+    proposedAppInstanceInfo.timeout.toNumber(),
```

Comment:
Why is it a BigNumber to begin with? My philosophy has been to keep things as `number` types as long as possible for things that rationally should be small numbers (e.g., the timeout or dependency nonce value) and as `BigNumber` the entire way through for things that rationally could be large numbers (e.g., the value in Wei of some free balance value).

Reply:
I kept this in parity with how an `AppInstanceInfo` is defined: https://github.com/counterfactual/monorepo/blob/master/packages/types/src/data-types.ts#L14